### PR TITLE
Fixes #235 add projectVersionPolicyId parameter to support VersionPolicy implementations

### LINKED ISSUE #235: Support Release Version Policies
gitflow-maven-plugin should support [VersionPolicy](https://maven.apache.org/maven-release/maven-release-api/apidocs/org/apache/maven/shared/release/policy/version/VersionPolicy.html) implementations that allows project to inject their own versioning strategy, the same way as maven-release-plugin does.

examples for such strategies:
* [OddEvenVersionPolicy](https://maven.apache.org/maven-release/maven-release-policies/maven-release-oddeven-policy/)
* [SemVerVersionPolicy](https://maven.apache.org/maven-release/maven-release-policies/maven-release-semver-policy/)

### PR DESCRIPTION
**Pull request by stefanseifert**
Fixes #235

### CODE REVIEW

**Comment by stefanseifert** [comment]
i've added an integration test

File: src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowVersionInfo.java [hunk 1]
```diff
@@ -59,6 +64,21 @@ public static boolean isValidVersion(final String version) {
                         .matcher(version).matches());
     }
 
+    @Override
+    public String getReleaseVersionString() {
+        if (versionPolicy != null) {
+            try {
+                VersionPolicyRequest request = new VersionPolicyRequest().setVersion(this.toString());
+                return versionPolicy.getReleaseVersion(request).getVersion();
+            } catch (PolicyException ex) {
```
Comment (aleksandr-m) [inline src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowVersionInfo.java:73]: Catch blocks can be merged together.

File: src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowVersionInfo.java [hunk 2]
```diff
@@ -90,6 +110,22 @@ public String nextSnapshotVersion(final Integer index) {
...
+        if (versionPolicy != null) {
+            try {
+                VersionPolicyRequest request = new VersionPolicyRequest().setVersion(this.toString());
+                if (snapshot) {
+                    return versionPolicy.getDevelopmentVersion(request).getVersion();
+                }
+                else {
+                    return versionPolicy.getReleaseVersion(request).getVersion();
+                }
+            } catch (PolicyException ex) {
```
Comment (aleksandr-m) [inline src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowVersionInfo.java:122]: Catch blocks can be merged together.

**Comment by aleksandr-m** [comment]
Thank you.

**Comment by stefanseifert** [comment]
great, thanks!

**Comment by aleksandr-m** [comment]
@stefanseifert I believe I've seen some review of the gitflow-maven-plugin written by you. Can you update it to reflect the latest changes?

**Comment by stefanseifert** [comment]
@aleksandr-m this PR is already merged (and released) - what do you mean?

**Comment by stefanseifert** [comment]
ah, i see. i've completely forgotten about this blog post, it's more a migration guide for users in the wcm.io ecosystem than a review. i've updated it to reflect the current feature set. i'm very happy with the current state of the gitflow plugin and appreciate all the work you've put into it in the last months!

**Comment by aleksandr-m** [comment]
Great. Thanks!

**Pull request merged**